models/data_model_pivot.go

```diff
@@ -77,16 +78,18 @@ func FieldFromPath(dm DataModel, pathLinkIds []string, baseTableName string) (Fi
 	linksMap := dm.AllLinksAsMap()
 	// check that the first link is from the base table
 	firstLink := linksMap[pathLinkIds[0]]
+	var fieldId string
 	if firstLink.ChildTableName != baseTableName {
 		return Field{}, errors.Wrap(
 			BadParameterError,
 			fmt.Sprintf(`first link's (%s) child table must be the base table "%s" (is "%s" instead)`,
 				firstLink.Id, baseTableName, firstLink.ChildTableName,
 			),
 		)
+	} else {
+		fieldId = firstLink.ParentFieldId
```

Comment:
this was an error (wasn't working for path of length one)

Reply:
What about this refactoring proposition ?
```go
// separate validation from field access, use it `data_model_usecase.go`
func ValidatePathPivot(dm DataModel, pathLinkIds []string, baseTableName string) error {
        linksMap := dm.AllLinksAsMap()
        
	// check that the first link is from the base table
	firstLink := linksMap[pathLinkIds[0]]
	if firstLink.ChildTableName != baseTableName {
		return Field{}, errors.Wrap(
			BadParameterError,
			fmt.Sprintf(`first link's (%s) child table must be the base table "%s" (is "%s" instead)`,
				firstLink.Id, baseTableName, firstLink.ChildTableName,
			),
		)
	}

	// check that the links are chained consistently
	for i := 1; i < len(pathLinkIds); i++ {
		previousLink := linksMap[pathLinkIds[i-1]]
		currentLink := linksMap[pathLinkIds[i]]
		if previousLink.ParentTableName != currentLink.ChildTableName {
			return Field{}, errors.Wrap(
				BadParameterError,
				fmt.Sprintf(`link %s (parent table "%s") is not a child of link %s (child table "%s")`,
					previousLink.Id, previousLink.ParentTableName, currentLink.Id, currentLink.ChildTableName,
				),
			)
		}
	}
}

// you can transform this method to return an error if len(pathLinkIds) == 0, but you get the idea
func FieldFromPath(dm DataModel, pathLinkIds []string, baseTableName string) Field {
        linksMap := dm.AllLinksAsMap()
        fieldId = linksMap[len(pathLinkIds)-1]
        return dm.AllFieldsAsMap()[fieldId]
}
```

An alternative is to keep both functions merged together (validation + access) but at least, separate validation from access (the current mix inside the loop for `fieldId` seems unecessary)

Reply:
Ok j'achète la proposition 👍